pptx_blueprint/__init__.py

```diff
@@ -54,7 +54,27 @@ def _find_shapes(self, label: str) -> Iterable[BaseShape]:
         Args:
             label (str): label of the placeholder (without curly braces)
         """
-        pass
+        slide_number, tag_name = label.split(":")
+        matched_shapes = []
+
+        def _find_shapes_in_slide(slide):
+            for shape in slide.shapes:
+                if shape.text == f'{{{tag_name}}}':
+                    matched_shapes.append(shape)
+
+        if slide_number == '*':
+            for slide in self._presentation.slides:
+                _find_shapes_in_slide(slide)
+        else:
+            # in label we are using 1 based indexing
+            slide_index = int(slide_number)-1
```

Comment:
why not extract these 3 lines in a separate private method named e.g. 'assert_valid_slide_index' that throws an exeption, or use the builtin 'assert', e.g. `assert slide_index_valid(slide_index)`. I think this would improve readability. And readability counts.